### PR TITLE
Upgrading to the latest oneDNN 2.7.1

### PR DESCRIPTION
**Pull request by ashiqimranintel**
Upgrading to the latest oneDNN 2.7.1

### CODE REVIEW

**Comment by ashiqimranintel** [comment]
@penpornk , can you approve this PR?

**Review by penpornk — APPROVED** [review]
Thank you for the update!

File: third_party/mkl_dnn/mkldnn_v1.BUILD [hunk 1]
```diff
@@ -116,7 +116,6 @@ _COPTS_LIST = select({
     "-UUSE_MKL",
     "-UUSE_CBLAS",
     "-DDNNL_ENABLE_MAX_CPU_ISA",
-    "-DDNNL_DISABLE_PRIMITIVE_CACHE",
```
Comment (penpornk) [inline third_party/mkl_dnn/mkldnn_v1.BUILD:119]: @ashiqimranintel This is intentional, right?

Comment (agramesh1) [reply third_party/mkl_dnn/mkldnn_v1.BUILD:119]: hi @penpornk yes.  This reverts a change in 2.10 https://github.com/tensorflow/tensorflow/pull/57071 due to a mem corruption that is now fixed.

Comment (penpornk) [reply third_party/mkl_dnn/mkldnn_v1.BUILD:119]: @agramesh1 Thank you for the quick reply!

**Comment by penpornk** [comment]
[ARM CI](https://github.com/tensorflow/tensorflow/actions/runs/3309074945/jobs/5461899463) failed, but it seems to be an existing failure since [another run](https://github.com/tensorflow/tensorflow/actions/runs/3308078945) before it (from another PR) also got the same failure.

**Comment by ashiqimranintel** [comment]
@penpornk , I think it's unrelated

**Pull request merged**